lib/ansible/modules/cloud/azure/azure_rm_availabilityset.py

```diff
@@ -207,33 +213,75 @@ def exec_module(self, **kwargs):
         # Check if the AS already present in the RG
         if self.state == 'present':
             response = self.get_availabilityset()
+            self.results['state'] = response
+
             if not response:
-                self.results['state'] = self.create_availabilityset()
-                self.results['changed'] = True
+                to_be_updated = True
             else:
-                self.log("AS already there, updating Tags")
                 update_tags, response['tags'] = self.update_tags(response['tags'])
+                
                 if update_tags:
-                    self.results['state'] = self.create_availabilityset()
-                    self.results['changed'] = True
+                    self.log("Tags has to be updated")
+                    to_be_updated = True
+                
+                if response['platform_update_domain_count'] != self.platform_update_domain_count :
+                    self.warn('platform_update_domain_count')
+                    self.platform_update_domain_count = response['platform_update_domain_count']
+
+                if response['platform_fault_domain_count'] != self.platform_fault_domain_count :
+                    self.warn('platform_fault_domain_count')
+                    self.platform_fault_domain_count = response['platform_fault_domain_count']
+
+                if response['sku'] != self.sku :
+                    self.warn('sku')
+                    self.sku = response['sku']
+
+            if self.warning:
+                self.module.warn("An Availability Set is immutable, except tags")
+
+            if self.check_mode:
+                return self.results
+
+            if to_be_updated:
+                self.results['state'] = self.create_or_update_availabilityset()
+                self.results['changed'] = True
+
         elif self.state == 'absent':
             self.delete_availabilityset()
             self.results['changed'] = True
 
         return self.results
 
-    def create_availabilityset(self):
+    def warn(self, param):
```

Comment:
Why is this a warning? IMO this should actually fail the module since most users don't read play output interactively.